qbg/ecp22.c

```diff
@@ -958,6 +958,7 @@ static int ecp22_req2send(char *ifname, unsigned short subtype,
 	ecp22_add_payload(ecp, payda);
 out:
 	LLDPAD_DBG("%s:%s rc:%d\n", __func__, ifname, rc);
+	free_pkd_tlv(ptlv);
```

Comment:
I don't believe this is the correct place to free this data. The ownership of this structure is passed to `payda` which is added to a list in the ecp data and remains in existence after this function. The TLV should only be freed in error return paths as it was originally. It was missing from some error pathways, but the current code is likely to lead to a crash now.

An alternative to removing this call to `free_pkg_tlv()`, you can set `ptlv` to `NULL` right after it's assigned to `payda->ptlv` on line 955 like this:
```
payda->ptlv = ptlv;
ptlv = NULL;
```
As I am not completely familiar with this area of the code, please report back if you believe this is in error and what you believe the actual lifetime of `ptlv` is.

Reply:
Yes, you are right, I should not call free_pkd_tlv() in the end if `rc == 0`. I think we can return before `out` if rc is 0.